app/javascript/flavours/glitch/components/status_prepend.js

```diff
@@ -5,36 +5,60 @@ import ImmutablePropTypes from 'react-immutable-proptypes';
 import { FormattedMessage } from 'react-intl';
 import Icon from 'flavours/glitch/components/icon';
 import { me } from 'flavours/glitch/initial_state';
+import NameList from './name_list';
 
 export default class StatusPrepend extends React.PureComponent {
 
   static propTypes = {
     type: PropTypes.string.isRequired,
-    account: ImmutablePropTypes.map.isRequired,
+    status: ImmutablePropTypes.map.isRequired,
+    accounts: ImmutablePropTypes.listOf(ImmutablePropTypes.map.isRequired),
     parseClick: PropTypes.func.isRequired,
     notificationId: PropTypes.number,
   };
 
-  handleClick = (e) => {
-    const { account, parseClick } = this.props;
-    parseClick(e, `/@${account.get('acct')}`);
+  handleAcctClick = (acct, e) => {
+    const { parseClick } = this.props;
+    parseClick(e, `/@${acct.get('acct')}`);
+  };
+
+  handleViewMoreClick = (e) => {
+    const { status, parseClick } = this.props;
+
+    const originalAuthor = status.getIn(['reblog', 'account', 'acct'], status.getIn(['account', 'acct']));
+    const originalStatusId = status.getIn(['reblog', 'id'], status.get('id'));
+    const url = `/@${originalAuthor}/${originalStatusId}` + this.getUrlSuffix();
+    parseClick(e, url);
+  };
```

Comment:
I do have another thing:
I don't think it's necessary to have and pass two different functions to NameList, which more or less do the same thing.
This could be one `handleClick` function, which just checks if `acct` is defined and changes the `url` passed to `parseClick`

Reply:
I was thinking this too while working on one of the other comments, I'll get this in 👍